modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/graphql/servlet/GraphQLServletExtender.java

```diff
@@ -1893,12 +1933,54 @@ private void _registerGraphQLDTOContributors(
 		GraphQLObjectType.Builder queryBuilder,
 		GraphQLSchema.Builder schemaBuilder) {
 
+		Map<String, GraphQLObjectType.Builder> queryNamespaceBuilderMap =
+			new HashMap<>();
+		Map<String, GraphQLObjectType.Builder> mutationNamespaceBuilderMap =
+			new HashMap<>();
+
 		for (GraphQLDTOContributor graphQLDTOContributor :
 				_graphQLDTOContributorServiceTrackerMap.values()) {
 
 			_registerGraphQLDTOContributor(
 				graphQLDTOContributor, mutationBuilder,
-				processingElementsContainer, queryBuilder, schemaBuilder);
+				processingElementsContainer, queryBuilder, schemaBuilder,
+				queryNamespaceBuilderMap, mutationNamespaceBuilderMap);
+		}
+
+		GraphQLCodeRegistry.Builder graphQLCodeRegistryBuilder =
+			processingElementsContainer.getCodeRegistryBuilder();
+
+		// Register GraphQLDTOContributor namespaces
+
+		for (Map.Entry<String, GraphQLObjectType.Builder> entry :
+				queryNamespaceBuilderMap.entrySet()) {
+
+			GraphQLObjectType.Builder queryNamespaceBuilder = entry.getValue();
+
+			queryBuilder.field(
+				_addField(queryNamespaceBuilder.build(), entry.getKey()));
+
+			schemaBuilder.codeRegistry(
+				graphQLCodeRegistryBuilder.dataFetcher(
+					FieldCoordinates.coordinates("query", entry.getKey()),
+					(DataFetcher<Object>)dataFetchingEnvironment -> new Object()
+				).build());
+		}
+
+		for (Map.Entry<String, GraphQLObjectType.Builder> entry :
+				mutationNamespaceBuilderMap.entrySet()) {
+
+			GraphQLObjectType.Builder mutationNamespaceBuilder =
+				entry.getValue();
+
+			mutationBuilder.field(
```

Comment:
this looks very similar to the other one...

Reply:
I'll generalize in a method to call the same logic from queries and mutations